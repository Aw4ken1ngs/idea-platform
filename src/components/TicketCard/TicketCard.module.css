.ticketCard {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 16px;
    border: 1px solid #ddd;
    border-radius: 8px;
    background-color: #fff;
    margin-bottom: 16px;
    margin-left: 8px;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
    font-family: Arial, sans-serif;
}

.ticketInfo {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-right: 16px;
}

.carrier {
    font-weight: bold;
    margin-bottom: 8px;
    font-size: 16px;
}

.buyButton {
    background-color: orange;
    color: white;
    border: none;
    border-radius: 4px;
    padding: 10px 14px;
    cursor: pointer;
    font-size: 14px;
    min-width: 200px;
}

.buyButton:hover {
    background-color: darkorange;
}

.departure,
.arrival {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 0 16px;
}

.time {
    font-size: 24px;
    font-weight: bold;
    margin-bottom: 4px;
}

.location {
    font-size: 16px;
    margin-bottom: 4px;
}

.date {
    font-size: 14px;
    color: #666;
}

.arrowBlock {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 0 16px;
}

.arrow {
    font-size: 24px;
    font-weight: bold;
    color: #888;
    margin-bottom: 4px;
}

.stops {
    font-size: 14px;
    color: #666;
}
