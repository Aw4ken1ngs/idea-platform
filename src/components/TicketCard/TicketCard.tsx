import React from 'react';
import { Currency, Ticket } from '../../types';
import { calculatePrice } from '../../utils/calculatePrice';
import CurrencyIcon from '../СurrencyIcon/CurrencyIcon';
import styles from './TicketCard.module.css';

type TicketCardProps = {
    ticket: Ticket;
    currency: Currency;
};

const TicketCard: React.FC<TicketCardProps> = ({ ticket, currency }) => {
    return (
        <div className={styles.ticketCard}>
            <div className={styles.ticketInfo}>
                <span className={styles.carrier}>{ticket.carrier}</span>
                <button className={styles.buyButton}>
                    Купить за {calculatePrice(ticket.price, currency)}{' '}
                    <CurrencyIcon currency={currency} />
                </button>
            </div>
            <div className={styles.departure}>
                <span className={styles.time}>{ticket.departure_time}</span>
                <span className={styles.location} data-testid={`location-${ticket.origin}`}>
                    {ticket.origin_name} ({ticket.origin})
                </span>
                <span className={styles.date}>{ticket.departure_date}</span>
            </div>
            <div className={styles.arrowBlock}>
                <span className={styles.arrow}>→</span>
                <span className={styles.stops}>{ticket.stops} пересадок</span>
            </div>
            <div className={styles.arrival}>
                <span className={styles.time}>{ticket.arrival_time}</span>
                <span className={styles.location}>
                    {ticket.destination_name} ({ticket.destination})
                </span>
                <span className={styles.date}>{ticket.arrival_date}</span>
            </div>
        </div>
    );
};

export default TicketCard;
