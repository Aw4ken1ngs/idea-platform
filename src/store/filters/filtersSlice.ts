import {createSlice} from "@reduxjs/toolkit";
import type {PayloadAction} from "@reduxjs/toolkit";
import {Currency} from "../../types";

interface FiltersState {
    currency: Currency;
    stops: number[];
}

const initialState:FiltersState = {
    currency: Currency.RUB,
    stops: []
}

export const filtersSlice = createSlice({
    name: 'filters',
    initialState,
    reducers: {
        setCurrency: (state, action: PayloadAction<Currency>) => {
            state.currency = action.payload;
        },
        setStops: (state, action: PayloadAction<number[]>) => {
            state.stops = action.payload;
        }
    }
})

export const { setCurrency, setStops } = filtersSlice.actions
export default filtersSlice.reducer