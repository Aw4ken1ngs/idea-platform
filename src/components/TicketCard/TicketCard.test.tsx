import { configureStore } from "@reduxjs/toolkit";
import { Currency } from "../../types";
import { Provider } from "react-redux";
import { render} from "@testing-library/react";
import * as filtersSlice from "../../store/filters/filtersSlice";
import TicketCard from "./TicketCard";
import ticketsData from "./../../tickets.json";
import '@testing-library/jest-dom';



const store = configureStore({
    reducer: {
        filters: filtersSlice.default,
    },
    preloadedState: {
        filters: {
            currency: Currency.RUB,
            stops: [],
        },
    },
});

const renderComponent = (ticket: typeof ticketsData.tickets[0]) => {
    return (
        <Provider store={store}>
            <TicketCard ticket={ticket} currency={Currency.RUB} />
        </Provider>
    );
};

describe('<TicketCard/>', () => {
    const mockTicket = ticketsData.tickets[0];

    it('should match snapshot', () => {
        const {container} = render(renderComponent(mockTicket));
        expect(container).toMatchSnapshot();
    });

    test('TicketCard renders correctly with mock data', () => {
        const {queryByTestId} = render(renderComponent(mockTicket));
        const element = queryByTestId('location-VVO');
        expect(element).toBeInTheDocument();
    });
})
