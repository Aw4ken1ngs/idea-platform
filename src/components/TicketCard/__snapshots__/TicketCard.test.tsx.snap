// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrencyFilter/> should match snapshot 1`] = `
<div>
  <div
    class="ticketCard"
  >
    <div
      class="ticketInfo"
    >
      <span
        class="carrier"
      >
        TK
      </span>
      <button
        class="buyButton"
      >
        Купить за 
        12400.00
         
        RUB
      </button>
    </div>
    <div
      class="departure"
    >
      <span
        class="time"
      >
        16:20
      </span>
      <span
        class="location"
        data-testid="location-VVO"
      >
        Владивосток
         (
        VVO
        )
      </span>
      <span
        class="date"
      >
        12.05.18
      </span>
    </div>
    <div
      class="arrowBlock"
    >
      <span
        class="arrow"
      >
        →
      </span>
      <span
        class="stops"
      >
        3
         пересадок
      </span>
    </div>
    <div
      class="arrival"
    >
      <span
        class="time"
      >
        22:10
      </span>
      <span
        class="location"
        data-testid="location-TLV"
      >
        Тель-Авив
         (
        TLV
        )
      </span>
      <span
        class="date"
      >
        12.05.18
      </span>
    </div>
  </div>
</div>
`;

exports[`<TicketCard /> should match snapshot 1`] = `
<div>
  <div
    class="ticketCard"
  >
    <div
      class="ticketInfo"
    >
      <span
        class="carrier"
      >
        Test Airline
      </span>
      <button
        class="buyButton"
      >
        Купить за 
        5000.00
         
        RUB
      </button>
    </div>
    <div
      class="departure"
    >
      <span
        class="time"
      >
        10:00
      </span>
      <span
        class="location"
      >
        Moscow
         (
        MOW
        )
      </span>
      <span
        class="date"
      >
        2025-01-10
      </span>
    </div>
    <div
      class="arrowBlock"
    >
      <span
        class="arrow"
      >
        →
      </span>
      <span
        class="stops"
      >
        0
         пересадок
      </span>
    </div>
    <div
      class="arrival"
    >
      <span
        class="time"
      >
        12:00
      </span>
      <span
        class="location"
      >
        Saint Petersburg
         (
        LED
        )
      </span>
      <span
        class="date"
      >
        2025-01-10
      </span>
    </div>
  </div>
</div>
`;
